### ISSUE TITLE
Added random_erasing and staged_lr

### ISSUE THREAD

**Issue by KaiyangZhou** [KaiyangZhou/deep-person-reid#110]
Updates:
1. [Random erasing data augmentation](https://github.com/zhunzhong07/Random-Erasing) can be enabled by adding argument `--augdata-re`.
2. You can use [staged learning rate](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/optimizers.py#L18), i.e. different learning rate for different layers. This can be managed by three arguments: (a) `--staged-lr`: called to use staged learning rate; (b) `--new-layers`: list of layer names (strings) indicating which layers use the default learning rate and the rest use a scaled learning rate; (c) `--base-lr-mult`: learning rate multiplier for base layers. For example, when you train `resnet50`, if you want the randomly initialized `self.classifier` to have `--lr 0.1` and the rest with learning rate scaled by 0.1, you can add `--staged-lr --new-layers classifier --base-lr-mult 0.1` to the argument list. See [here](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/optimizers.py#L23) for more details.

**Comment by KaiyangZhou** [comment]
Flag to call random erasing is changed to `--random-erase`.

**Comment by KaiyangZhou** [comment]
New features in [transforms.py](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/transforms.py#L125):

`--color-jiter`: randomly change the brightness, contrast and saturation

`--color-aug`: randomly alter the intensities of RGB channels
